packages/vitest/src/node/pools/pool.ts

```diff
@@ -167,28 +167,30 @@ export class Pool {
   }
 
   async cancel(): Promise<void> {
-    // Set flag to prevent new tasks from being queued
-    this._isCancelling = true
```

Comment:
Pool should already have had all tests queued at this point. Or if there was another group, the `ctx.isCancelling` in `node/pool.ts` should have prevented them from being queued.